analyzer/compliance-analyzer/src/main/java/eu/fasten/analyzer/complianceanalyzer/Main.java

```diff
@@ -0,0 +1,62 @@
+package eu.fasten.analyzer.complianceanalyzer;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.sql.SQLException;
+
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import eu.fasten.server.connectors.PostgresConnector;
+import picocli.CommandLine;
+
+
+@CommandLine.Command(name = "ComplianceAnalyzer")
+public class Main implements Runnable{
+
+    private static Logger logger = LoggerFactory.getLogger(Main.class);
+
+    @CommandLine.Option(names = {"-f", "--file"},
+        paramLabel = "JSON",
+        description = "Path to JSON file which contains the repo info")
+    String jsonFile;
+
+    @CommandLine.Option(names = {"-d", "--database"},
+        paramLabel = "dbURL",
+        description = "Database URL for connection",
+        defaultValue = "jdbc:postgresql:postgres")
+    String dbUrl;
+
+    @CommandLine.Option(names = {"-u", "--user"},
+        paramLabel = "dbUser",
+        description = "Database user name",
+        defaultValue = "postgres")
+    String dbUser;
+    public static void main(String[] args) {
+      final int exitCode = new CommandLine(new Main()).execute(args);
+      System.exit(exitCode);
+    }
+
+    @Override
+    public void run() {
+      var compliancePlugin = new ComplianceAnalyzerPlugin.CompliancePluginExtension();
+      /*try {
+        compliancePlugin.setDBConnection(PostgresConnector.getDSLContext(dbUrl, dbUser));
+      } catch (IllegalArgumentException | SQLException e) {
+        logger.error("Could not connect to the database", e);
+        return;
+      }*/
```

Comment:
Why commented out?

Reply:
I guess this was commented to point out that in the long term our plugin should consume a Kafka Topic to retrieve repo information. As long as it is not available, we will simulate this retrieval by reading from a [dummy JSON file](https://github.com/fasten-project/fasten/blob/endocode/compliancePlugin/analyzer/compliance-analyzer/dummyKafkaTopic.json).

This is now better documented:
https://github.com/fasten-project/fasten/blob/af7c6e4c8453777e377953a1de6fb821059ec779/analyzer/compliance-analyzer/src/main/java/eu/fasten/analyzer/complianceanalyzer/Main.java#L60-L94

Reply:
Fair enough, but it has nothing to do with PostgresConnector. We mostly don't use Main class to consume from Kafka, it's for debugging purposes that is run on single coordinates or sets of coordinates.

Reply:
Ok. Could you please show me where exactly it is best to consume the Kafka topic?

Reply:
Normally the FASTEN server consumes messages from Kafka topic and then passes them to the `consume` method of the plugin (in this case ComplianceAnalyzerPlugin), then it calls `produce` to get the output. So you shouldn't care about consuming messages yourself. Those messages will be provided to the plugin by the server when you run it with the server. On the other hand, if you run it through Main (which is mostly used for debugging), then you shouldn't consume messages from Kafka at all, you just pass coordinates yourself. That's how it's handled in all other FASTEN plugins

Reply:
> Normally the FASTEN server consumes messages from Kafka topic and then passes them to the `consume` method of the plugin (in this case ComplianceAnalyzerPlugin), then it calls `produce` to get the output. So you shouldn't care about consuming messages yourself. Those messages will be provided to the plugin by the server when you run it with the server. 

Got it.

> On the other hand, if you run it through Main (which is mostly used for debugging), then you shouldn't consume messages from Kafka at all, you just pass coordinates yourself. That's how it's handled in all other FASTEN plugins

I don't think I got this. By "coordinates" you mean the Kafka topic's URI?

As far as our demo is concerned, can we still simulate the consumption of such a topic by reading a file? What changes should I make to the code?

Reply:
By "coordinates" I meant any input the plugin needs. I've been mostly working with plugins that use Maven coordinates (groupID:artifactID:version) as their input, so now I accidentally call every input a coordinate. 
With regard to the demo you could make a file with multiple messages and read it from the Main class. So you'll need some parameter for the path to such a file that your Main class will take.

Reply:
> So you'll need some parameter for the path to such a file that your Main class will take.

We do already have a Maven argument for that:
https://github.com/fasten-project/fasten/blob/f9943e41b20ee7be6728358644f922cc67a9b344/analyzer/compliance-analyzer/src/main/java/eu/fasten/analyzer/complianceanalyzer/Main.java#L36-L40

Is that enough?